randutil/random.go

```diff
@@ -22,80 +19,89 @@ func init() {
 }
 
 // Salt generates a new random salt of the given size.
-func Salt(size int) ([]byte, error) {
+func Salt(size int) []byte {
 	salt := make([]byte, size)
-	_, err := io.ReadFull(rand.Reader, salt)
-	if err != nil {
-		return nil, errors.Wrap(err, "error generating salt")
+	// crypto/rand.Reader always fill the entire slices.
+	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
+	// This error check can be removed once we drop support for Go 1.23.
```

Comment:
```suggestion
	// crypto/rand.Reader always fills the entire slice.
	// Since Go 1.24, it is guaranteed to never return an error, see https://github.com/golang/go/issues/66821.
	// This error check can be removed once we drop support for Go 1.23.
```

Reply:
GitHub broke the "Commit suggestion" UI, so I fixed it manually in 17c0809136a9c61cf660b3f7fc2640586a23afdd.